Mathlib/Combinatorics/SimpleGraph/Matching.lean

```diff
@@ -133,6 +133,48 @@ lemma IsMatching.coeSubgraph {G' : Subgraph G} {M : Subgraph G'.coe} (hM : M.IsM
   · obtain ⟨_, hw', hvw⟩ := (coeSubgraph_adj _ _ _).mp hy
     rw [← hw.2 ⟨y, hw'⟩ hvw]
 
+lemma IsMatching.iff_map_equiv {G' : SimpleGraph W} {M : Subgraph G} (f : SimpleGraph.Iso G G') :
+    M.IsMatching ↔ (M.map f.toHom).IsMatching := by
```

Comment:
Can you first show that matchings can be pushforwarded along bijective graph homs?
```suggestion
lemma Iso.isMatching_map {G' : SimpleGraph W} {M : Subgraph G} (f : SimpleGraph.Iso G G') :
    (M.map f.toHom).IsMatching ↔ M.IsMatching := by
```

Reply:
By this you mean the edit you suggested, right? In that case it was the case that the name mismatched the contents, and I've fixed that:)

Reply:
No, a graph isomorphism is stronger than a bijective graph homomorphism. I rather mean something like
```
protected lemma IsMatching.map {G' : SimpleGraph W} {M : Subgraph G} (f : G →g G') (hf : Bijective f) :
    M.IsMatching → (M.map f.toHom).IsMatching := by
```
Then you can prove `Iso.isMatching_map` by two applications of `IsMatching.map`

Reply:
It took a while, but I finally got to it:)